task_02/tevfik_aksoy/main.cpp

```diff
@@ -0,0 +1,128 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <tuple>
+#include <set>
+#include <fstream>
+
+using std::cout;
+using std::endl;
+using std::set;
+using std::tuple;
+using std::vector;
+
+//Structure of nodes(vertices)
+struct Node
+{
+	int node_id = 0;
+	Node()
+	{
+		node_id = 0;
+	}
+	Node(int node_id)
+	{
+		this->node_id = node_id;
+	}
+};
+
+//Tuple of <int,int> for realization of vector of vectors of <int,int>
+typedef tuple<int, int> DoubleTuple;
+
+//Structure of the edge including the source and destination nodes in it
+struct Edge
+{
+	Node src_node = {};
+	Node dest_node = {};
+	int edge_val = 0;
+	Edge(Node src_node, Node dest_node, int edge_val)
+	{
+		this->src_node = src_node;
+		this->dest_node = dest_node;
+		this->edge_val = edge_val;
+	}
+};
+
+class Graph
+{
+public:
+	vector<vector<DoubleTuple>> graphVec;
+	Graph(vector<Edge> &edges, int node_amount)
+	{
+		graphVec.resize(node_amount);
+		for (int i = 0; i < edges.size(); i++)

```

Comment:
В данном случае лучше использовать запись цикла:
```suggestion
              for (const auto& edge : edges) {
```